### PR TITLE
Add loss_utils for rewriting lce_forward methods

### PR DESCRIPTION
**Pull request by Tcc0403**
## Summary

This PR introduces a loss_utils module which includes a series of functions for loss calculations.
I think developers can benefit from this PR for the following reasons:
1. Code structure is more similar to huggingface/transformers, which can make developers easier to support new models with flce.
2. Better readability for tracking the related changes in transformers and updating.

## Testing Done

- Hardware Type: <BLANK>
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

### CODE REVIEW

File: src/liger_kernel/transformers/model/loss_utils.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    # Shift so that token < n predict n
+    labels = nn.functional.pad(labels, (0, 1), value=ignore_index)
+    shift_labels = labels[..., 1:].contiguous()
+
+    # Flatten the tokens
+    hidden_states = hidden_states.view(-1, hidden_size)
+    shift_labels = shift_labels.view(-1)
+    # Enable model parallelism
+    shift_labels = shift_labels.to(hidden_states.device)
+    loss = fixed_fused_lienar_cross_entropy(
```
Comment (shivam15s) [inline src/liger_kernel/transformers/model/loss_utils.py:49]: there's a typo in linear

File: src/liger_kernel/transformers/model/gemma2.py [hunk 2]
```diff
@@ -212,6 +213,14 @@ def lce_forward(
...
     # if in training mode, don't materialize logits
     if self.training and (labels is not None):
+        loss = LigerForCausalLMLoss(
+            hidden_states=hidden_states,
+            lm_head_weight=self.lm_head.weight,
+            labels=labels,
+            hidden_size=self.config.hidden_size,
+            softcap=self.config.final_logit_softcapping,
+            **loss_kwargs,
+        )
```
Comment (shivam15s) [inline src/liger_kernel/transformers/model/gemma2.py:223]: seems like you missed deleting the old implementation here?

**Review by shivam15s — CHANGES_REQUESTED** [review]

**Review by shivam15s — APPROVED** [review]
lgtm, the failing bwd test is unrelated
